homeassistant/components/elmax/common.py

```diff
@@ -137,21 +147,25 @@ def __init__(
         coordinator: ElmaxCoordinator,
     ) -> None:
         """Construct the object."""
+        super().__init__(coordinator=coordinator)
         self._panel = panel
         self._device = elmax_device
         self._panel_version = panel_version
-        self._coordinator = coordinator
-        self._transitory_state = None
+        self._client = coordinator.http_client
+        self._transitory_state: Any = None
+        # Keep a local reference to the coordinator to better deal with type checking ad avoid
+        # casting.
+        self._elmax_coordinator = coordinator
```

Comment:
We don't need this, as we already have `self.coordinator` provided by the `CoordinatorEntity`

Reply:
Hi @frenck : I am well aware of that. 
However, I tought (as the comment states) this would be better than having to cast the coordinator in all subclasses when I need to use it.

This allows me to avoid casts as it would be in the following example (without that reference)
![image](https://user-images.githubusercontent.com/4648843/145281890-dcd7d23d-5842-4cbd-b6a7-99dd6ef6b374.png)

Do you prefer the "casting" approach? Maybe I'm missing something here: if you have any more coincise and elegant way to do so, feel free to suggest!

Reply:
You don't need to cast it, see my above code example added.

Reply:
Ok, I see it!